### PR TITLE
Do not allow additional properties in most values in order to avoid unnoticed typos

### PR DESCRIPTION
**Pull request by AndiDog**
### What this PR does / why we need it

I stumbled over `global.connectivity.eniModePodSubnets` vs. `global.connectivity.network.eniModePodSubnets` by mistake. Customers may also step into such issues, wondering why default/different values are used. Instead, a hard error is more appropriate if a key just doesn't exist (e.g. mistyped).

### Checklist

- [x] Updated CHANGELOG.md.

### Trigger E2E tests

/run cluster-test-suites

### CODE REVIEW

**Review by Gacko — APPROVED** [review]

**Review by fiunchinho — APPROVED** [review]

**Review by nprokopic — APPROVED** [review]

**Comment by AndiDog** [comment]
Seems this doesn't work – I'll park this and test it at some time later, since it's not critical right now.

**Comment by fiunchinho** [comment]
There is this field called `awsAccountId` that we were using only to render the template in CI. That value was missing from the schema. I pushed a commit to see if that helps.

**Comment by fiunchinho** [comment]
This is needed https://github.com/giantswarm/cluster-standup-teardown/pull/89

File: helm/cluster-aws/values.schema.json [hunk 1]
```diff
@@ -1682,6 +1692,11 @@
                             "title": "Amazon machine image (AMI)",
                             "description": "If specified, this image will be used to provision EC2 instances."
                         },
+                        "awsAccountId": {
+                            "type": "string",
+                            "title": "AWS Account id",
```
Comment (Gacko) [inline helm/cluster-aws/values.schema.json:1697]:
```suggestion
                            "title": "AWS Account ID",
```

File: helm/cluster-aws/README.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ Properties within the `.global.providerSpecific` object
 | `global.providerSpecific.additionalResourceTags` | **Additional resource tags** - Additional tags to add to AWS resources created by the cluster.|**Type:** `object`<br/>|
 | `global.providerSpecific.additionalResourceTags.*` | **Tag value**|**Type:** `string`<br/>**Value pattern:** `^[ a-zA-Z0-9\._:/=+-@]+$`<br/>|
 | `global.providerSpecific.ami` | **Amazon machine image (AMI)** - If specified, this image will be used to provision EC2 instances.|**Type:** `string`<br/>|
+| `global.providerSpecific.awsAccountId` | **AWS Account id** - Only used when rendering the chart template locally, you shouldn't use this value. Used to calculate the IRSA service account issuer when using the China region.|**Type:** `string`<br/>|
```
Comment (Gacko) [inline helm/cluster-aws/README.md:23]:
```suggestion
| `global.providerSpecific.awsAccountId` | **AWS Account ID** - Only used when rendering the chart template locally, you shouldn't use this value. Used to calculate the IRSA service account issuer when using the China region.|**Type:** `string`<br/>|
```

**Comment by fiunchinho** [comment]
/run cluster-test-suites TARGET_SUITES=./providers/capa/china

**Comment by fiunchinho** [comment]
Adding `skip/ci` label because tests have passed.

**Pull request merged**